lib/resty/acme/storage/redis.lua

```diff
@@ -55,8 +57,31 @@ local function op(self, op, ...)
   return ok, err
 end
 
+local function add_namespace(namespace, key)
+  if namespace == "" then
+    return key
+  else
+    -- <namespace>::<real_key>
+    return fmt("%s::%s", namespace, key)
```

Comment:
I would actually suggest to not hardcode the namespace pattern here, instead, just treat it as a prefix.
Then our logic can be simplified a lot, for get/set we can blindly do `key = namespace .. key`

Reply:
@fffonion Do you mean let the upper level users do this logic? Sounds more reasonable.

Reply:
yes

Reply:
Hi, I'm on the Kong SRE team and we are looking to use namespaces. :) 

Can a single `:` be used to separate the namespace from the key? [This](https://redis.com/blog/5-key-takeaways-for-developing-with-redis/) Redis blog post mentions that convention:

> When you namespace your keys by application or service – the convention is to use the colon (‘:’) character to delimit parts of the key’s name -a Redis namespace best practice.

Reply:
@szesch Thanks for the nice suggestion. I feel that, namespace is a special part of the key, different from other parts. Using `::` can make this special part apparent and reduce the likelihood of confusion.
For example, we have a key `foo:bar` stored in redis. It could either be namespace `foo` + real_key `bar`, or empty namespace + real_key `foo:bar`. @fffonion How do you think?

Reply:
The pattern of namespace should be left to the user, i.e. we simply concat namespace .. key in our library.